lib/icingadb/icingadb-objects.cpp

```diff
@@ -2589,6 +2593,14 @@ void IcingaDB::StateChangeHandler(const ConfigObject::Ptr& object, const CheckRe
 	}
 }
 
+void IcingaDB::ReachabilityChangeHandler(const Checkable::Ptr& checkable)
+{
+	for (const IcingaDB::Ptr& rw : ConfigType::GetObjectsByType<IcingaDB>()) {
+		rw->UpdateState(checkable);
+		rw->SendStatusUpdate(checkable);
+	}
+}
```

Comment:
This is fine for the `is_reachable` attribute, but `severity` of a service actually depends on the reachability of its host: https://github.com/Icinga/icinga2/blob/5017f5fa439f7a21d5201800faa618cc912d429b/lib/icinga/service.cpp#L140-L141

So to update the current severity correctly, if `checkable` is a host, we'd have to update that state of all its services.

I think I discussed with @lippserd that it might also be a valid fix to just make the service severity independent of the host reachability. Maybe @nilmerg also has an opinion here as the primary use of severity is to use it as sort order in web. We can implement both, keeping the current severity just implies that we have to send more updates.

Reply:
Is this still relevant? Anyhow, I think the host's state and reachable state is indeed important in the severity calculation. For the same reason we include the downtime depth and acknowledge state in it. Though, we don't include dependencies do we? These fall in the same area to me as the immediate parent host itself. But I'm not asking to consider dependencies now, I just want to show that as long as we don't do that, it's a half-baked solution anyway. But one that exists a very long time now, so I'd be surprised to see a service (that is critical because it's host is down/unreachable) to be equally weighted as any other critical service in the sort order of my incident list. So my reason for keeping the current state is just: It's been always the case this way.

Reply:
Nice reasoning. xD

Reply:
> Though, we don't include dependencies do we? These fall in the same area to me as the immediate parent host itself. But I'm not asking to consider dependencies now, I just want to show that as long as we don't do that

`IsReachable()` corresponds to dependencies, but we only consider dependencies of the host for the severity of a service (and it only has an effect if the host is still considered UP). Both hosts and services are not affected by their own dependencies at the moment.

> it's a half-baked solution anyway.

Agreed.